app/Domains/Contact/ManageContact/Web/Controllers/ContactController.php

```diff
@@ -24,10 +25,26 @@ class ContactController extends Controller
 {
     public function index(Request $request, Vault $vault)
     {
-        $contacts = Contact::where('vault_id', $vault->id)
-            ->where('listed', true)
-            ->orderBy('created_at', 'asc')
-            ->paginate(25);
+        switch (Auth::user()->contact_sort_order) {
+            case User::CONTACT_SORT_ORDER_ASC:
+                $contacts = Contact::where('vault_id', $vault->id)
+                    ->where('listed', true)
+                    ->orderBy('last_name', 'asc')
+                    ->paginate(25);
+                break;
+            case User::CONTACT_SORT_ORDER_DESC:
+                $contacts = Contact::where('vault_id', $vault->id)
+                    ->where('listed', true)
+                    ->orderBy('last_name', 'desc')
+                    ->paginate(25);
+                break;
+            default:
+                $contacts = Contact::where('vault_id', $vault->id)
+                ->where('listed', true)
+                ->orderBy('last_updated_at', 'desc')
+                ->paginate(25);
+                break;
+        }
```

Comment:
Is it possible to not duplicate the whole query each time?

```suggestion
            $contacts = Contact::where('vault_id', $vault->id)
                    ->where('listed', true);

            switch (Auth::user()->contact_sort_order) {
            case User::CONTACT_SORT_ORDER_ASC:
                $contacts = $contacts->orderBy('last_name', 'asc');
                break;
            case User::CONTACT_SORT_ORDER_DESC:
                $contacts = $contacts->orderBy('last_name', 'desc');
                break;
            default:
                $contacts = $contacts->orderBy('last_updated_at', 'desc');
                break;
        }
        
        $contacts = $contacts->paginate(25);
```

Reply:
done